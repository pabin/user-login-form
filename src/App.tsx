import React from 'react';
import './App.css';
import Heading from './components/Heading';
import Form from './components/Form';

function App() {
  return (
    <div className="App">
      <Form />
      <Heading />
    </div>
  );
}

export default App;
