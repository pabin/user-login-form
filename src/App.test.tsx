import React from 'react';
import { render } from '@testing-library/react';
import App from './App';


describe('App', () => {
  test('renders app component correctly', () => {
    const wrapper = render(<App />);
    expect(wrapper.baseElement).toMatchSnapshot();
  });
});

