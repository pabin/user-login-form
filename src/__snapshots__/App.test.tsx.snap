// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders app component correctly 1`] = `
<body>
  <div>
    <div
      class="App"
    >
      <div
        class="container"
      >
        <span
          class="stepCount"
          data-testid="currentStep"
        >
          Step 
          1
           of 3 . . .
        </span>
        <div
          class="formContainer"
        >
          <p
            class="title"
          >
            Let's set up your account
          </p>
          <p>
            Already have an account?
             
            <a
              href="#login"
            >
              Sign in
            </a>
          </p>
          <div
            class="form"
          >
            <div
              class="container"
            >
              <fieldset
                class="fieldset"
              >
                <legend
                  class="legend"
                  data-testid="legend-Your Name"
                />
                <input
                  class="input"
                  data-testid="Your Name"
                  placeholder="Your Name"
                  type="text"
                />
              </fieldset>
            </div>
            <div
              class="container"
            >
              <fieldset
                class="fieldset"
              >
                <legend
                  class="legend"
                  data-testid="legend-Email Address"
                />
                <input
                  class="input"
                  data-testid="Email Address"
                  placeholder="Email Address"
                  type="email"
                />
              </fieldset>
            </div>
            <div
              class="container"
            >
              <select
                class="select"
                data-testid="form-select"
                id="cars"
                name="cars"
              >
                <option
                  value=""
                >
                  I would describe my user type as
                </option>
                <option
                  value="Developer"
                >
                  Developer
                </option>
                <option
                  value="QA Engineer"
                >
                  QA Engineer
                </option>
                <option
                  value="System Admin"
                >
                  System Admin
                </option>
              </select>
            </div>
            <div
              class="container"
            >
              <fieldset
                class="fieldset"
              >
                <legend
                  class="legend"
                  data-testid="legend-Password"
                />
                <input
                  class="input"
                  data-testid="Password"
                  placeholder="Password"
                  type="password"
                />
                <img
                  alt="eye"
                  class="icon"
                  data-testid="eye"
                  height="18"
                  src="eye.png"
                  width="18"
                />
              </fieldset>
              <span
                class="helpText"
                data-testid="passwordHelpText"
              >
                Minimum 8 Characters
              </span>
            </div>
            <button
              class="button"
              disabled=""
            >
              Next
            </button>
          </div>
          <p
            class="footerText"
          >
            By clicking the "Next" button, you agree to creating a free account, and to
             
            <a
              href="#terms-od-service"
            >
              Terms of Service
            </a>
             and 
            <a
              href="#privacy-policy"
            >
              Privacy Policy
            </a>
          </p>
        </div>
      </div>
      <div
        class="container"
      >
        <p
          class="title"
        >
          Dummy Heading
        </p>
        <p
          class="description"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore at dolore magna aliqua
        </p>
      </div>
    </div>
  </div>
</body>
`;
